src/browser-installer/chromium/utils.ts

```diff
@@ -0,0 +1,36 @@
+import os from "os";
+import path from "path";
+import { BrowserPlatform } from "@puppeteer/browsers";
+import { getChromePlatform, getMilestone } from "../utils";
+import { CHROMEDRIVER_STORAGE_API, MIN_CHROMEDRIVER_MAC_ARM_NEW_ARCHIVE_NAME } from "../constants";
+
+export const getChromiumBuildId = async (platform: BrowserPlatform, milestone: string | number): Promise<string> => {
+    const { default: revisions } = await import(`./revisions/${platform}`);
+
+    return String(revisions[milestone]);
+};
+
+export const getChromeDriverArchiveUrl = (version: string): string => {
+    const chromeDriverArchiveName: Record<BrowserPlatform, string> = {
+        linux: "linux64",
+        mac: "mac64",
+        mac_arm: "mac64_m1", // eslint-disable-line camelcase
+        win32: "win32",
+        win64: "win32",
+    };
+
+    const milestone = getMilestone(version);
+    const platform = getChromePlatform(version);
+    const isNewMacArm =
+        platform === BrowserPlatform.MAC_ARM && Number(milestone) >= MIN_CHROMEDRIVER_MAC_ARM_NEW_ARCHIVE_NAME;
+    const archiveName = isNewMacArm ? "mac_arm64" : chromeDriverArchiveName[platform];
```

Comment:
Starting at chromium@106, chromium driver started shipping mac arm builds with "mac_arm64" prefix instead of old "mac64_m1"